examples/mgpcg_advanced.py

```diff
@@ -138,24 +177,24 @@ def apply_preconditioner(self):
                 self.smooth(l, 1)
                 self.smooth(l, 0)
 
-    @ti.kernel
-    def paint(self):
-        if ti.static(self.dim == 3):
-            kk = self.N_tot * 3 // 8
-            for i, j in self.pixels:
-                ii = int(i * self.N / self.N_gui) + self.N_ext
-                jj = int(j * self.N / self.N_gui) + self.N_ext
-                self.pixels[i, j] = self.x[ii, jj, kk] / self.N_tot
+    def solve(self, steps=-1, eps=1e-12, tol=1e-12, rel=1e-12):
```

Comment:
`step` should probably be renamed to `max_iters`. `tol` and `rel` seem to be `abs_tolerance` and `rel_tolerance`? Please rename accordingly.